### PR TITLE
[Codebase]: GitHub Action: Publishing to NPM

### PR DESCRIPTION
**Pull request by Fritsch-Tech**
#### 🔧 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Codebase improvement
- [ ] Other (if none of the other choices apply)

#### 🚨 Checklist

- [x] I've read the [guidelines for contributing](https://github.com/gremo/react-directus/blob/main/CONTRIBUTING.md)
- [x] I've added necessary documentation (if appropriate)
- [x] I've ensured that my code additions do not fail linting or unit tests (if applicable)

#### Description

This pull request addresses issue:
 - #380

I have added a GitHub Action for publishing to NPM. The action requires the `NPM_AUTH_TOKEN`, which will be used for authentication during the publishing process. To ensure security, I have not included the token directly in the workflow file. Instead, you will need to generate an NPM token and add it to your repository's secrets (Settings -> Secrets). The action will then access the token from the secrets and use it for authentication to NPM.

Please review the changes and let me know if there are any other requirements or adjustments needed.

Thank you for considering this pull request, and I look forward to your feedback on the GitHub Action implementation.

### CODE REVIEW

**Comment by gremo** [comment]
This is not easy as it sounds. While the action could work, it is still too "manual". I mean, when you locally run npm version minor, two things happen:

- a git tag is created
- a version commit is created, with the default message same as the tag

(I hope to recall correctly, documentation here: https://docs.npmjs.com/cli/v9/commands/npm-version#description)

The action I linked expect the message in the format "Release x.y.z". So, in order to work, one (that is me, actually) should change the commit message (automatically created) every time, then push.

Maybe we should change the defualt "Release x.y.z" pattern matching or, even better, hook into the tag push event and work directly with the tagged version.

I may be wrong, what's your opinion?

**Comment by Fritsch-Tech** [comment]
I just added `message="Release %s"` to the `.npmrc` file.
Now the commit message created by `npm version` has the correct format.

File: .github/workflows/npm-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    - name: Run build
+      run: npm run build
+      
+    - name: Publish if version has been updated
+      uses: pascalgn/npm-publish-action@1.3.9
+      with: 
+        tag_name: "v%s"
+        tag_message: "v%s"
+        create_tag: "true"
```
Comment (gremo) [inline .github/workflows/npm-publish.yaml]: There is no need to create a tag

File: .github/workflows/npm-publish.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        
+    - name: Run clean
+      run: npm run clean
+
+    - name: Run build
+      run: npm run build
+      
+    - name: Publish if version has been updated
+      uses: pascalgn/npm-publish-action@1.3.9
+      with:
```
Comment (gremo) [inline .github/workflows/npm-publish.yaml]: All default options can be removed

**Comment by gremo** [comment]
> I just added `message="Release %s"` to the `.npmrc` file. Now the commit message created by `npm version` has the correct format.

Seems nice to solve that specif problem.

However I prefer a "push on tag" approach. There is no need to install and build on every commit!

I'll do some experiements as soon as I have some free time, in the meanwhile let's put this in standby waiting for a better solution 👍 

Thanks anyways!

File: .github/workflows/npm-publish.yaml (hunk 1 above)
Comment (Fritsch-Tech) [reply .github/workflows/npm-publish.yaml]: i set create_tag to false.

File: .github/workflows/npm-publish.yaml (hunk 2 above)
Comment (Fritsch-Tech) [reply .github/workflows/npm-publish.yaml]: Which ones do you mean?

**Comment by Fritsch-Tech** [comment]
I added 
```
on:
  push:
    branches:
      - main
    tags:
      - 'v*.*.*'
```

Was that what you ment?

**Comment by gremo** [comment]
> I added
> 
> ```
> on:
>   push:
>     branches:
>       - main
>     tags:
>       - 'v*.*.*'
> ```
> 
> Was that what you ment?

Yes.

But then, what's the point of using the action? We know that the workflow is triggered on tag now. The tag is created automatically with npm version, locally. So the only think we need to do is to publish, without the need to parse Release (%s) or use an external action.

Like https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

Do you agree?

**Comment by Fritsch-Tech** [comment]
Yes, so basicly like this

**Comment by gremo** [comment]
I've merged this but, ideally, we want just 1 action to lint, test, build (on push/pr) and (eventually, if it's a tag) publish to npm.

**Pull request merged**